extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -246,26 +261,152 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
+        const viewportSize = this.getViewport().getSize();
+        return this.attemptChangeSize(
+          AmpAdNetworkAdsenseImpl.getResponsiveHeightForContext_(
+            this.autoFormat_,
+            viewportSize,
+            this.element
+          ),
+          viewportSize.width
+        ).catch(() => {});
+      }
+
+      // This should happen last, as some diversion criteria rely on some of the
+      // preceding logic (specifically responsive logic).
+      this.divertExperiments();
+      this.maybeAddSinglePassExperiment();
+    });
+  }
+
+  /**
+   * Upgrades the ad unit to responsive if there is an opt-in setting in localstorage.
+   * See https://github.com/ampproject/amphtml/issues/23568 for design.
+   * @return {Promise} a promise that resolves when any upgrade is complete.
+   */
+  maybeUpgradeToResponsive() {
+    // If the ad unit is already responsive, there's nothing to do.
+    // Note that we can't use this.autoFormat_ here because it's not
+    // populated yet.
+    if (this.element.hasAttribute('data-auto-format')) {
+      return Promise.resolve();
+    }
+    return (
+      Services.storageForDoc(this.element)
+        .then(storage => storage.get(this.getAdSizeOptimizationStorageKey()))
+        .then(isAdSizeOptimizationEnabled => {
+          if (isAdSizeOptimizationEnabled) {
+            this.upgradeToResponsive();
+          }
+        })
+        // Do nothing if we fail to read localstorage.
+        .catch(() => {
+          dev().warn(TAG, 'Failed to look up publisher ad size settings.');
+        })
+    );
+  }
+
+  /**
+   * Actually upgrades the ad unit to responsive.
+   */
+  upgradeToResponsive() {
+    this.element.setAttribute('height', ADSENSE_RSPV_WHITELISTED_HEIGHT);
+    this.element.setAttribute('width', '100vw');
+    this.element.setAttribute('data-full-width', '');
+    this.element.setAttribute('data-auto-format', 'rspv');
+
+    devAssert(this.isValidElement(), 'Upgraded element is not valid.');
+  }
+
+  /**
+   * Sets up a listener for a pingback of publisher settings in the ad response and
+   * writing such settings to localstorage.
+   *
+   * Note that we can have multiple listeners on the same page, which is okay because
+   * the settings for publishers should not change between different slots.
+   *
+   * See https://github.com/ampproject/amphtml/issues/23568 for design.
+   */
+  attachSettingsListener() {
+    const listener = event => {
+      if (event['source'] != this.iframe.contentWindow) {
+        return;
+      }
+      const data = getData(event);
+      // data will look like this:
+      // {
+      //   'googMsgType': 'adsense-settings',
+      //   'adClient': 'ca-pub-123',
+      //   'enableAutoAdSize': true
+      // }
+      if (!!data && data['googMsgType'] != 'adsense-settings') {
+        return;
+      }
+      if (data['adClient'] != this.getAdClientId()) {
+        return;
+      }
+      const autoAdSizeStatus = !!data['enableAutoAdSize'];
```

Comment:
if you truly expect it to the boolean true then better to do:  const autoAdSizeStatus = data['enableAutoAdSize'] === true;

Reply:
Looking at some ad response, it seems it's actually '1' - updated.